presto-main/src/main/java/com/facebook/presto/operator/aggregation/histogram/Histogram.java

```diff
@@ -125,16 +126,15 @@ public static void input(Type type, HistogramState state, Block key, int positio
 
     public static void combine(HistogramState state, HistogramState otherState)
     {
-        if (state.get() != null && otherState.get() != null) {
-            TypedHistogram typedHistogram = state.get();
-            long startSize = typedHistogram.getEstimatedSize();
-            typedHistogram.addAll(otherState.get());
-            state.addMemoryUsage(typedHistogram.getEstimatedSize() - startSize);
-        }
-        else if (state.get() == null && otherState.get() != null) {
-            TypedHistogram otherTypedHistogram = otherState.get();
-            state.set(otherTypedHistogram);
-        }
+        // NOTE: state = current merged state; otherState = scratchState (new data to be added)
+        // for grouped histograms and single histograms, we have a single histogram object. In neither case, can this otherState.get() return null.
+        // Semantically, a histogram object will be returned even the group is empty. In that case, the histogram object will represent an empty histogram until we call add() on
+        // it.
```

Comment:
nit: merge this `it` to the previous line :P

Reply:
i thought I went over our 180 limit
i actually re-wrote the comment slightly so it definitely goes over. I've reformatted this comment so it's not so lop-sided as well